internal/controller/apiextensions/claim/reconciler.go

```diff
@@ -342,12 +344,13 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		"version", cm.GetResourceVersion(),
 		"external-name", meta.GetExternalName(cm),
 	)
+	cnd := cnds.New(&cm.Unstructured, cnds.WithRecorder(r.record))
 
 	// Check the pause annotation and return if it has the value "true" after
 	// logging, publishing an event and updating the Synced status condition.
 	if meta.IsPaused(cm) {
-		r.record.Event(cm, event.Normal(reasonPaused, reconcilePausedMsg))
-		cm.SetConditions(xpv1.ReconcilePaused().WithMessage(reconcilePausedMsg))
+		cnd.SetCondition(cnds.Normal(xpv1.ReconcilePaused().WithMessage(reconcilePausedMsg)))
```

Comment:
example of using the proposed condition approach that will only emit events if the condition has changed